src/main/java/com/powsybl/openloadflow/network/util/GenerationActionPowerDistributionStep.java

```diff
@@ -92,4 +105,19 @@ public double run(List<ParticipatingElement> participatingElements, int iteratio
 
         return done;
     }
+
+    private double getParticipationFactor(LfGenerator generator) {
+        double factor;
+        switch (participationType) {
+            case MAX:
+                factor = generator.getMaxP() / generator.getDroop();
+                break;
+            case TARGET:
+                factor = generator.getTargetP();
```

Comment:
Finally, I think that the absolute value is required here.

Reply:
Okay thanks I will modify the computation.

Reply:
Can you confirm it's only for the factor in targetP mode ?